### PR TITLE
[FIF-152] Implement Add student note

### PR DESCRIPTION
**Pull request by NSD-Xavier**

### CODE REVIEW

File: EdFi.Buzz.UI.Angular/src/app/Features/StudentDetail/studentDetail.component.ts [hunk 1]
```diff
@@ -69,14 +70,18 @@ export class StudentDetailComponent implements OnInit {
...
+      }) ;
   }
   cancelAddNote() {
     this.editingNote = -1;
     this.student.notes.splice(0, 1);
   }
-  deleteNote(id: number) {
-    const idx = this.student.notes.findIndex(el => el.id === id);
+  deleteNote(studentNoteKey: number) {
+    const idx = this.student.notes.findIndex(el => el.studentnotekey === studentNoteKey);
```
Comment (lalrg) [inline EdFi.Buzz.UI.Angular/src/app/Features/StudentDetail/studentDetail.component.ts:84]: We still need a new story for deleting notes.

File: EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts [hunk 2]
```diff
@@ -80,12 +78,12 @@ export class StudentApiService {
       .sort((a, b) => a.name.localeCompare(b.name));
   }
 
-  public async getById(id: string) {
-    console.log('student.service:' + id);
+  public async getById(studentSchoolKey: string) {
+    console.log('student.service:' + studentSchoolKey);
```
Comment (lalrg) [inline EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts:82]: I think we should avoid including console.logs in the functionalities, at least in most of them.

File: EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts [hunk 3]
```diff
@@ -80,12 +78,12 @@ export class StudentApiService {
...
     let student: Student;
 
     const client = this.apollo.getClient();
-    const queryparams = { staffkey: this.auth.currentUserValue.teacher.staffkey, studentschoolkey: id };
+    const queryparams = { staffkey: this.auth.currentUserValue.teacher.staffkey, studentschoolkey: studentSchoolKey };
```
Comment (lalrg) [inline EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts:86]:
```suggestion
    const queryParams = { staffkey: this.auth.currentUserValue.teacher.staffkey, studentschoolkey: studentSchoolKey };
```

File: EdFi.Buzz.UI.Angular/src/app/Features/StudentDetail/studentDetail.component.ts (hunk 1 above)
Comment (NSD-Xavier) [reply EdFi.Buzz.UI.Angular/src/app/Features/StudentDetail/studentDetail.component.ts:84]: Deleting notes to back end is not implemented,its just the local delete that was implemented some time ago

File: EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts (hunk 2 above)
Comment (NSD-Xavier) [reply EdFi.Buzz.UI.Angular/src/app/Services/student.service.ts:82]: I agree

**Pull request merged**